src/youtube2zim/utils.py

```diff
@@ -31,12 +31,29 @@ def load_json(cache_dir, key):
     if not fname.exists():
         return None
     try:
-        with open(fname, "r") as fp:
+        with open(fname) as fp:
             return json.load(fp)
     except Exception:
         return None
 
 
+def load_mandatory_json(cache_dir, key):
+    """load mandatory JSON collection from path or raise an error"""
+    fname = cache_dir.joinpath(f"{key}.json")
+    if not fname.exists():
```

Comment:
read_xxx() as well. WDYT of dropping the if exists? I think the read will raise a close-enough exception should it not exist and we're already not taking care of permissions or i/o errors

Reply:
Yes indeed